libs/openFrameworks/gl/ofGLProgrammableRenderer.cpp

```diff
@@ -1315,12 +1321,46 @@ void ofGLProgrammableRenderer::bind(const ofFbo & fbo, bool setupPerspective){
 }
 
 //----------------------------------------------------------
-void ofGLProgrammableRenderer::unbind(const ofFbo & fbo){
-	glBindFramebuffer(GL_FRAMEBUFFER, 0);
+void ofGLProgrammableRenderer::end(const ofFbo & fbo){
+	fbo.unbind();
```

Comment:
hey, yeah sorry. the idea is that if you wanted to use OF with several gl contexts from different threads, now you can do it as long as you don't use:
```
someglobject.draw(..)
```

or

```
someglobject.bind()
```

which are not thread safe because they call the ofGetCurrentRenderer which is a singleton and will fail if called from multiple threads simultaneously. instead now you can do:

```
renderer.draw(someglobject)
```

and

```
renderer.bind(someglobject)
```

so all the draw and bind methods in a renderer should not access any singleton even indirectly. if a renderer calls object.bind, object.unbind or object.draw from any of it's methods then it breaks the thread safety.

there's an example in apps/devApps/explicitRendererExample that doesn't use threads but uses this alternative syntax to bind and draw gl objects through a renderer instead of directly through the object.